src/config.cpp

```diff
@@ -160,14 +161,19 @@ void CConfig::commence() {
 }
 
 static std::expected<int64_t, std::string> configStringToInt(const std::string& VALUE) {
+    auto parseHex = [](const std::string& value) -> std::expected<int64_t, std::string> {
+        try {
+            size_t position;
+            auto   result = stoll(value, &position, 16);
+            if (position == value.size())
+                return result;
+        } catch (std::out_of_range) { return std::unexpected("hex " + value + " is too big"); } catch (...) {
```

Comment:
why are there two catches here

Reply:
My idea was that a huge hex is kind of valid, so we need to handle it separately from an invalid/malformed hex.
Should it also return invalid ?

Reply:
you are catching twice, there are two catches

also `catch (std::exception&)`

Reply:
It will catch only once, example: 
```
try {
    throw std::out_of_range("FFF");
} catch (std::out_of_range) {
    std::cout << "out of range\n";
} catch (...) {
    std::cout << "other\n";
}
```

https://en.cppreference.com/w/cpp/string/basic_string/stol
First catch is for `out_of_range`, the second catch is for `invalid_argument`.

Reply:
ye but we dont need this, just catch `std::exception&`

Reply:
yeah well now you dont print any error, just throw a generic "error in parsing"

Reply:
Now I'm returning `"invalid hex " + value`, should I change it to `"cannot parse \"" + value + "\" as n hex."` ?

Reply:
ah right my bad I looked wrong